src/EnrollDuoController.js

```diff
@@ -38,9 +38,11 @@ function (Okta, Duo, Q, FormController, Footer) {
 
       getInitOptions: function () {
         return this.doTransaction(function (transaction) {
-          return transaction
-          .getFactorByTypeAndProvider('web', 'DUO')
-          .enrollFactor();
+          var factor = _.findWhere(transaction.factors, {
+            factorType: 'web',
+            provider: 'DUO'
+          });
+          return factor.enroll();
```

Comment:
Does a signature like this improve maintainability?:

    return FactorUtil.enrollFactor({
      factors: transaction.factors,
      provider: 'OKTA',
      factorType: self.get('factorType'),
      data: {
        profile: profileData
      }
    });

Another option is:

    return FactorUtil.enrollFactor(transaction.factors, 'OKTA', self.get('factorType'), {
      profile: profileData
    });